cfpb/consumerfinance.gov
# Mega Menu: Check that relatedTarget is not null in TabTrigger

https://github.com/cfpb/consumerfinance.gov/pull/6719 refactored the tab trigger around the `focusout` event. However, in Safari on desktop and iOS the `focusout` event has a `relatedTarget` that is `null`, which gets around checks to see if the tabbing is happening outside of the menu. This triggers a collapse on the menu when the menu back button is pressed in Safari on desktop and iOS. This PR checks if `relatedTarget` is `null` and if it is, it bails out of the `focusout` handler.

## Changes

- Check if `event.relatedTarget === null`, and bail out of `focusout` event if so.


## How to test this PR

1. In Safari, you should be able to navigate to the first and second level of the menu and back. You should also be able to tab and shift tab through the menu.
2. In iOS Safari, you should be able to navigate to the first and second level of the menu and back.
3. FF, Chrome, and IE should work as they work on production.

## Notes
- It's possible to tab to hidden elements of the menu in IE11 because `relatedTarget` is erroneously null and bails out. Unfortunately, the solution in https://stackoverflow.com/questions/41298948/how-to-use-relatedtarget-or-equivalent-in-ie would break in Safari.
